File: 1669-merge-in-between-linked-lists/1669-merge-in-between-linked-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeInBetween(self, list1: ListNode, a: int, b: int, list2: ListNode) -> ListNode:
        dummy = ListNode(-1, list1)
        previous = dummy
        slow = list1
        fast = list1
        diff = b - a

        while diff:
            fast = fast.next
            diff -= 1
        

        while fast and a:
            previous = slow
            slow = slow.next
            fast = fast.next

            a -= 1
        
        print(slow.val, fast.val, previous.val)
        previous.next = list2

        node = list2
        while node.next:
            node = node.next
        
        if fast and fast.next:
            node.next = fast.next

        return dummy.next

File: 1669-merge-in-between-linked-lists/1669-merge-in-between-linked-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {number} a
 * @param {number} b
 * @param {ListNode} list2
 * @return {ListNode}
 */

var mergeInBetween = function(list1, a, b, list2) {
    let counter = 0;
    let list2Tail = null
    
    let node = list2;
    while (1) {
        if (node.next)
            list2Tail = node.next;    
        else 
            break
        
        node = node.next
    }
    
    node = list1;
    while(node) {
        if(counter === b)  {
            list2Tail.next=node.next;
            node.next = null
            break
        }
                
        counter++;
        node = node.next;
    }
    
    counter = 1;
    node = list1;
    while(node) {
        if(counter === a)  {
            node.next=list2;
            break
        }
                
        counter++;
        node = node.next;
    }
      
    return list1;
};